awsUtilities/src/main/java/org/opensearch/migrations/aws/SigV4Signer.java

```diff
@@ -61,76 +48,67 @@ public SigV4Signer(
         this.region = region;
         this.protocol = protocol;
         this.timestampSupplier = timestampSupplier;
+        this.bodyStream = null;
     }
 
-    public void consumeNextPayloadPart(ByteBuffer payloadChunk) {
-        if (payloadChunk.remaining() <= 0) {
-            return;
+    private static void writeByteBufferToByteArrayOutputStream(ByteBuffer byteBuffer,
+                                                               ByteArrayOutputStream outputStream) {
+        var readBuffer = byteBuffer.duplicate();
+
+        // If the ByteBuffer has a backing array, use it directly
+        if (readBuffer.hasArray()) {
+            outputStream.write(readBuffer.array(), readBuffer.position(), readBuffer.remaining());
+        } else {
+            // Otherwise, read bytes manually into a temporary array
+            byte[] temp = new byte[readBuffer.remaining()];
+            readBuffer.get(temp);
+            outputStream.write(temp, 0, temp.length);
         }
-        if (messageDigest == null) {
-            try {
-                this.messageDigest = MessageDigest.getInstance("SHA-256");
-            } catch (NoSuchAlgorithmException e) {
-                throw Lombok.sneakyThrow(e);
-            }
+    }
```

Comment:
This code is far inferior to the old code. We may have multiple 100MB requests going through the replayer at a time.  All that you need here is a checksum, so there's no reason to buffer hundreds of MBs of data and put the whole process at risk of crashing or thrashing.  Please change this.  There's a reason that the API and implementation were constructed as they were before.